backend/common/src/main/java/ai/verta/modeldb/common/CommonUtils.java

```diff
@@ -174,6 +178,64 @@ public static <T extends GeneratedMessageV3> void observeError(
     responseObserver.onError(logError(e));
   }
 
+  public static <T> void observeError(CompletableFuture<T> completableFuture, Throwable e) {
+    completableFuture.completeExceptionally(observeErrorWithResponse(e, e.getMessage(), LOGGER));
+  }
+
+  public static ResponseStatusException observeErrorWithResponse(
```

Comment:
There's a lot of code in here. We should have unit tests for this.

Reply:
We do not have a unit test for a modeldb backend so I do not have it for this but we have many Integration tests which used this code and validate exceptions.

Reply:
but, you could definitely add unit tests for this code, correct?

Reply:
Added unit test for this code.